src/eval/mod.rs

```diff
@@ -1077,7 +1086,14 @@ impl Eval for ast::FuncCall {
             } else {
                 let target = target.eval(vm)?;
                 let args = args.eval(vm)?;
-                if !matches!(target, Value::Symbol(_) | Value::Module(_)) {
+
+                // Prioritize a function's own methods (with, where) over its fields.
```

Comment:
no need to duplicate the comment, bad enough that the whole code is a bit duplicate here (not your fault)